src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -0,0 +1,173 @@
+            // Instead, ensure program correctness without tracing.
+            try
+            {
+                if (s_controllerInstance == null)
+                {
+                    string strEnabledValue = CompatibilitySwitch.GetValueInternal("EnableEventPipe");
+                    if (strEnabledValue != null)
+                    {
+                        int enabledValue = Convert.ToInt32(strEnabledValue);
+                        if (enabledValue == 4)
+                        {
+                            s_controllerInstance = new EventPipeController();
+                        }
+                    }
+                }
+            }
+            catch { }
+        }
+
+        private EventPipeController()
+        {
+            // Determine the marker file path.
+            string traceFilePath = CompatibilitySwitch.GetValueInternal("EventPipeOutputFile");
+            if (!string.IsNullOrEmpty(traceFilePath))
+            {
+                m_traceFilePath = traceFilePath;
+                m_markerFilePath = traceFilePath + MarkerFileExtension;
+            }
+            else
+            {
+                // If the output file path is not specified then use
+                // path/to/current/working/directory/Process-<pid>.netperf.ctl
+                m_traceFilePath = EventPipeInternal.GetDefaultTraceFileName();
+                m_markerFilePath = m_traceFilePath + MarkerFileExtension;
+            }
+
+            m_markerFileExists = false;
+
+            // Start a new thread to listen for tracing commands.
+            Task.Factory.StartNew(WaitForTracingCommand, TaskCreationOptions.LongRunning);
+        }
+
+        private void WaitForTracingCommand()
+        {
+            while (true)
+            {
+                // Make sure that any transient errors don't cause the listener thread to exit.
+                try
+                {
+                    // Check for existence of the file.
+                    // If the existence of the file has changed since the last time we checked
+                    // this means that we need to act on that change.
+                    bool fileExists = File.Exists(m_markerFilePath);
+                    if (m_markerFileExists != fileExists)
+                    {
+                        // Save the result.
+                        m_markerFileExists = fileExists;
+
+                        // Take the appropriate action.
+                        if (fileExists)
+                        {
+                            // Enable tracing.
+                            EventPipe.Enable(GetConfiguration());
+                        }
+                        else
+                        {
+                            // Disable tracing.
+                            EventPipe.Disable();
+                        }
+                    }
+
+                    // Wait for the polling interval.
+                    Thread.Sleep(PollingIntervalMilliseconds);
```

Comment:
While polling every 10 seconds to enable makes sense to keep the overhead very low, it does not make sense for disable .   It would be useful to change to to 1 sec when enabled like this.  
```
 Thread.Sleep(PollingIntervalMilliseconds);
```
to 
```
 Thread.Sleep(isEnabled ? 1 : PollingIntervalMilliseconds);
```

Reply:
Agreed.  Fixed.